### DISCUSSION TITLE
Host a website without Domain?

### DISCUSSION THREAD

**Discussion by blueYOSHI9000** [community/community#21947] [Other Feature Feedback, Questions, & Ideas]
<p>First off, I’m very new here so I’m sorry if this is the wrong place to ask!</p>
<p>Hi, I just recently started on a small website project and decided to upload it to Github. I already saw that ypu can host website on here and already found a tutorial. However, apparently I need a domain for that, isn’t there a way without? What I mean is just the normal github domain /myproject or something, is that possible?</p>

**Comment by mpboom** [comment, marked as answer]
<p>First of all, welcome to the community! Secondly, it this might have been a better place to post: <a href="/t5/Pages/bd-p/pages">https://github.community/t5/Pages/bd-p/pages</a>, since I think the feature you are referring to in your post is Github Pages. Anyway, I’ll try to answer your question.</p>
<p>You have to types of GH Pages sites: an orginization or user site - this is a website for your whole profile and a repository stie - this is a site intended for use by just one repository.</p>
<p>If you want to create an orginization/user site just create a repository named <a href="http://yourUserName.github.io" rel="nofollow noopener">yourUserName.github.io</a>. Then, follow the tutorials on how to set up a GH Pages site in that repository.</p>
<p>If you want to create a site for a single repository, just create a branch named “gh-pages” and put your site in there, according to the tutorials.</p>
<p>If you want to access your user/orginazation site just head over to <a href="http://yourUserName.github.io" rel="nofollow noopener">yourUserName.github.io</a>. If you want to access a repository site just head over to <a href="http://yourUserName.github.io/respositoryName" class="inline-onebox-loading" rel="nofollow noopener">yourUserName.github.io/respositoryName</a>!</p>
<p>You CAN youse a custom domain if you want to, but it is not mandatory.</p>
<p>The following link acutally contains all info on what I just said: <a href="https://help.github.com/articles/user-organization-and-project-pages/" rel="nofollow noopener">https://help.github.com/articles/user-organization-and-project-pages/</a>.</p>
<p>I hope I anwered your question.</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/blueyoshi9000">@blueyoshi9000</a>,</p>
<p>As <a class="mention" href="/mpboom">@mpboom</a> mentioned, this post really belongs in the Pages board. I went ahead and moved it for you. This means you’ll get better engagement, and it keeps our Community organized so users can more easily find information.</p>
<p>No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Let me know if you have any other questions or if we can help with anything else.</p>
<p>Cheers!</p>

**Comment by jameswrightbrown-prog** [comment]
Yes, it’s actually possible to host a website without buying a domain. GitHub provides GitHub Pages, which automatically gives you a free subdomain like `username.github.io/project-name`. When I was working on a client project related to[ emirates appliances ](https://emiratesappliances.com/)we first tested and hosted the site using GitHub Pages without purchasing a domain. It worked perfectly for previewing and sharing the project. You just need to enable GitHub Pages in the repository settings and select the branch (usually `main`). After that, your site becomes accessible through the GitHub URL.
